sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/NuGetMetadataReference.cs

```diff
@@ -109,110 +92,146 @@ private static MetadataReference[] CreateReferences(string packageId, string pac
 
-        internal static Dictionary<string, MetadataReference[]> NUnit =>
-            lazyNUnit.Value;
+        public static MetadataReference[] MicrosoftAspNetCoreRoutingAbstractions(string packageVersion) =>
+            Create("Microsoft.AspNetCore.Routing.Abstractions", packageVersion);
 
-        internal static Dictionary<string, MetadataReference[]> SystemCollectionsImmutable =>
-            lazySystemCollectionsImmutable.Value;
+        public static MetadataReference[] MicrosoftAspNetMvc(string packageVersion) =>
+            Create("Microsoft.AspNet.Mvc", packageVersion);
 
-        internal static Dictionary<string, MetadataReference[]> SystemThreadingTasksExtensions =>
-            lazySystemThreadingTasksExtensions.Value;
+        public static MetadataReference[] MSTestTestFrameworkV1 =>
+            Create("MSTest.TestFramework", "1.1.11");
 
-        internal static Dictionary<string, MetadataReference[]> XunitAssert =>
-            lazyXunitAssert.Value;
+        public static MetadataReference[] MSTestTestFramework(string packageVersion) =>
+            Create("MSTest.TestFramework", packageVersion);
 
-        internal static Dictionary<string, MetadataReference[]> XunitExtensibilityCore =>
-            lazyXunitExtensibilityCore.Value;
+        public static MetadataReference[] NUnit(string packageVersion) =>
+            Create("NUnit", packageVersion);
 
-        internal static MetadataReference[] XunitV1Packages =>
-            lazyXunitV1.Value.Values.First()
-            .Concat(lazyXunitExtensionsV1.Value.Values.First())
-            .ToArray();
+        public static MetadataReference[] SystemCollectionsImmutable(string packageVersion) =>
+            Create("System.Collections.Immutable", packageVersion);
 
-        [AssemblyInitialize]
-        public static void SetupAssembly(TestContext context)
-        {
-            // Choosing one day to reduce the waiting time when a new version of the used nugets is
-            // released. If the waiting time when running tests locally is big we can increase.
-            const int VersionCheckDelayInDays = 1;
+        public static MetadataReference[] SystemThreadingTasksExtensions(string packageVersion) =>
+            Create("System.Threading.Tasks.Extensions", packageVersion);
 
-            // Install new nugets only once per day to improve the performance when running tests locally
-            // When adding a new nuget it is recommended to delete the content of
-            // sonar -csharp\sonaranalyzer-dotnet\TestResults
-            if (DateTime.Now.Subtract(TestResultHelper.GetPreviousRunDate(context)).TotalDays < VersionCheckDelayInDays)
-            {
-                return;
-            }
+        public static MetadataReference[] XunitFramework(string packageVersion) =>
+            Create("xunit.assert", packageVersion)
+            .Concat(Create("xunit.extensibility.core", packageVersion))
+            .ToArray();
```

Comment:
This function now returns all of the references required by Xunit, so the caller doesn't need to specify each package separately like before.